io-engine/src/rebuild/rebuild_job.rs

```diff
@@ -164,9 +165,19 @@ impl RebuildJob {
 
     /// Schedules the job to start in a future and returns a complete channel
     /// which can be waited on.
-    pub fn start(
+    pub async fn start(
         &self,
+        map: Option<RebuildMap>,
     ) -> Result<oneshot::Receiver<RebuildState>, RebuildError> {
+        let (s, r) = oneshot::channel();
+        self.comms
+            .send(RebuildJobRequest::SetRebuildMap((map, s)))
+            .await
+            .ok();
+        if let Err(e) = r.await {
+            error!("{uri}: failed to set rebuild map: {e}", uri = self.dst_uri);
```

Comment:
could be worth making it more explicit saying a full rebuild will be done? though this should never happen I guess..

Reply:
If it cannot communicate to backend, it's broken already.